### PR TITLE
Running local testnet from celotool

### PR DESCRIPTION
**Pull request by H34D**
### Description

- [x] spin up network with validator nodes

### Related issues

- Solves one aspect of https://github.com/celo-org/celo-monorepo/issues/1807

### CODE REVIEW

**Comment by nategraf** [comment]
FYI: I've also been working on https://github.com/celo-org/celo-monorepo/pull/1907, mostly for my own benefit, but maybe some of the code would be useful here.

**Comment by H34D** [comment]
> FYI: I've also been working on #1907, mostly for my own benefit, but maybe some of the code would be useful here.

amazing, thanks

**Comment by H34D** [comment]
ci/circleci: contractkit-test is fixed in https://github.com/celo-org/celo-monorepo/pull/2150

**Comment by H34D** [comment]
ci/circleci: protocol-test is fixed in https://github.com/celo-org/celo-monorepo/pull/2155

File: packages/celotool/package.json [hunk 1]
```diff
@@ -35,20 +35,24 @@
   },
   "devDependencies": {
     "@types/bunyan": "1.8.4",
+    "@types/chai": "^4.1.3",
     "@types/dotenv": "^4.0.3",
+    "@types/mocha": "^5.2.1",
     "@types/node-fetch": "^2.1.2",
+    "@types/portscanner": "^2.1.0",
```
Comment (nategraf) [inline packages/celotool/package.json:42]: Why was this version downgraded?

File: packages/celotool/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,979 @@
+{
```
Comment (nategraf) [inline packages/celotool/package-lock.json]: Why was this file added?

File: packages/celotool/package.json [hunk 3]
```diff
@@ -35,20 +35,24 @@
...
     "@types/prompts": "^1.1.1",
     "@types/web3": "^1.0.18",
     "@types/yargs": "^13.0.2",
-    "@types/mocha": "^5.2.1",
-    "@types/chai": "^4.1.3",
     "chai": "^4.1.2",
-    "mocha": "^5.1.1"
+    "commander": "^2.20.3",
```
Comment (nategraf) [inline packages/celotool/package.json:47]: Why is `commander` being added? Is that use case not covered by `yargs` and `prompts`?

File: packages/celotool/register.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (nategraf) [inline packages/celotool/register.sh:1]: Are you intending to check this in? If so, it'd be much better to have this part of a command in Typescript rather than as a shell script.

File: packages/celotool/register_validatorgroup.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
```
Comment (nategraf) [inline packages/celotool/register_validatorgroup.sh:1]: Same comment as `packages/celotool/register.sh`

File: packages/celotool/src/e2e-tests/utils.ts [hunk 6]
```diff
@@ -1,45 +1,36 @@
+import { consoleLogger } from '@celo/utils/lib/logger'
```
Comment (nategraf) [inline packages/celotool/src/e2e-tests/utils.ts]: I don't see a good reason to import this. It's just an alias for `console.log`
https://github.com/celo-org/celo-monorepo/blob/master/packages/utils/src/logger.ts#L15

**Review by nategraf — COMMENTED** [review]
Sorry I haven't had time to do a complete review yet. Here are some initial comments.

File: packages/celotool/package-lock.json (hunk 2 above)
Comment (H34D) [reply packages/celotool/package-lock.json]: adding it was unintended, removed.

File: packages/celotool/package.json (hunk 1 above)
Comment (H34D) [reply packages/celotool/package.json:42]: This is the latest version of the portscanner types. 

we are using 
    `"portscanner": "^2.2.0"`
and 
    `"@types/portscanner": "^2.1.0",`

File: packages/celotool/src/e2e-tests/utils.ts (hunk 6 above)
Comment (H34D) [reply packages/celotool/src/e2e-tests/utils.ts]: ok, removed

**Comment by H34D** [comment]
given up for #2264
